helm-chart/kuberay-operator/templates/deployment.yaml

```diff
@@ -37,11 +37,22 @@ spec:
           command:
             - /manager
           args:
-          {{- if .Values.batchScheduler.enabled }}
-            - --enable-batch-scheduler
-          {{- else }}
-            []
-          {{- end }}
+            {{- $argList := list -}}
+            {{- if .Values.batchScheduler.enabled -}}
+            {{- $argList = append $argList "--enable-batch-scheduler" -}}
+            {{- end -}}
+            {{- $watchNamespace := "" -}}
+            {{- if .Values.singleNamespaceInstall -}}
+            {{- $watchNamespace = .Release.Namespace -}}
+            {{- else if .Values.watchNamespace -}}
+            {{- $watchNamespace = .Values.watchNamespace -}}
+            {{- end -}}
+            {{- if $watchNamespace -}}
+            {{- $argList = append $argList "--watch-namespace" -}}
+            {{- $argList = append $argList $watchNamespace -}}
+            {{- end -}}
+            {{- (printf "\n") -}}
```

Comment:
I am not familiar with the `printf` function in Helm. Would you mind explaining the purpose of `{{- (printf "\n") -}}` here?

Reply:
It prints a newline to ensure the arg list is rendered as a valid yaml list (starting on the line after the key)

Reply:
Thank you for the explanation!